docs/standard/base-types/backtracking-in-regular-expressions.md

```diff
@@ -116,7 +123,7 @@ If you do not set a time-out value explicitly, the default time-out value is det
  By default, the time-out interval is set to <xref:System.Text.RegularExpressions.Regex.InfiniteMatchTimeout?displayProperty=nameWithType> and the regular expression engine does not time out.
 
 > [!IMPORTANT]
-> We recommend that you always set a time-out interval if your regular expression relies on backtracking.
+> We recommend that you always set a time-out interval if your regular expression relies on backtracking or operates on untrusted inputs.
```

Comment:
We did the work to make timeouts work in the NonBacktracking engine. If nothing else, they protect against vast inputs. Why not recommend them?

Reply:
indeed, if you're using this engine it's probably because you value predictable performance over the utmost performance, so the overhead of timeout checks is likely to be acceptable.

Reply:
> If nothing else, they protect against vast inputs

No, they don't. For example, there's no timeout checking while finding the next possible matching location in the backtracking engines, there's no timeout checking in the 2nd or 3rd phases of matching in the non-backtracking engine, etc. Timeouts are only there to prevent excessive backtracking.

Reply:
> Timeouts are only there to prevent excessive backtracking.

why did we implement timeout checks in the non backtracking engine?

Reply:
As a nod to the fact that timeouts exist. And they're only in one phase. Basically is it easier to explain that they don't exist at all or that they're checked sporadically... we decided the latter.

Reply:
I'll add the qualification you suggest, but this seems a wierd situation, given they will cause the regex to time out in at least some situations, which presumably is better than none.. not sure why we want to explicitly not recommend them there.

Reply:
It's not that we're saying "don't use them". We're only saying "you need to use them" in some situations. There's a difference.